### PR TITLE
Homework Pizzas #3

### PR DESCRIPTION
**Pull request by DamiXioR**

### CODE REVIEW

**Review by MaciejSalamonski — APPROVED** [review]
code looks good 👍

File: test/PizzeriaTest.cpp [hunk 1]
```diff
@@ -65,3 +71,35 @@ TEST_F(PizzeriaTest, calculatePriceForPizzaMock)
...
+    constexpr double cheaperPizza = 10.99;
+    constexpr double moreExpensivePizza = 250.0;
+
+    NiceMock<PizzaMock> niceMockPizza{};
+    StrictMock<PizzaMock> strictMockPizza{};
+
+    Pizzeria bravo("Bravo Pizza", mockTimer);
+    Pizzas pizzas = {new PizzaStub("Stub"), new PizzaStub("Stub2"), new PizzaStub("Stub3"), &niceMockPizza, &strictMockPizza};
+
+    EXPECT_CALL(mockTimer, sleep_for(minutes(1))).Times(3);
```
Comment (jakjan95) [inline test/PizzeriaTest.cpp:89]: avoid magic values for time :)

**Review by jakjan95 — COMMENTED** [review]
Just a few magic values, but code Looks good :)

File: test/PizzeriaTest.cpp (hunk 1 above)
Comment (WBigaj) [reply test/PizzeriaTest.cpp:89]: +1

**Review by WBigaj — COMMENTED** [review]
I have checked your code and left my feedback.

File: test/PizzeriaTest.cpp (hunk 1 above)
Comment (DamiXioR) [reply test/PizzeriaTest.cpp:89]: just tests :)

File: test/PizzeriaTest.cpp [hunk 2]
```diff
@@ -1,19 +1,23 @@
...
+struct PizzeriaTest : public ::testing::Test {
 public:
-    Pizzeria pizzeria = Pizzeria("dummyName"); 
+    StrictMock<MockTimer> mockTimer;
+    Pizzeria pizzeria = Pizzeria("dummyName", mockTimer);
 };
 
+struct PizzeriaTestMain : public ::testing::Test {
+    StrictMock<MockTimer> mockTimer;
+};
```
Comment (ziobron) [inline test/PizzeriaTest.cpp:20]: No need to have separate Fixture class. You could use PizzeriaTest.

File: test/PizzeriaTest.cpp (hunk 1 above)
Comment (ziobron) [reply test/PizzeriaTest.cpp:89]: Yup, avoid magic values in tests as well. What is minutes(1)? You should rather have variable
`minutes stubBakingTime = minutes(1);` and the test will be more readable.

File: test/PizzeriaTest.cpp [hunk 3]
```diff
@@ -65,3 +71,35 @@ TEST_F(PizzeriaTest, calculatePriceForPizzaMock)
...
+    EXPECT_CALL(niceMockPizza, getBakingTime()).WillOnce(Return(margheritaBakingTime));
+    EXPECT_CALL(mockTimer, sleep_for(margheritaBakingTime)).Times(1);
+
+    EXPECT_CALL(strictMockPizza, getPrice()).WillOnce(Return(moreExpensivePizza));
+    EXPECT_CALL(strictMockPizza, getName()).WillOnce(Return("Special Funghi With Olives"));
+    EXPECT_CALL(strictMockPizza, getBakingTime()).WillOnce(Return(funghiBakingTime));
+    EXPECT_CALL(mockTimer, sleep_for(funghiBakingTime)).Times(1);
+
+    auto orderId = bravo.makeOrder(pizzas);
+    auto price = bravo.calculatePrice(orderId);
```
Comment (ziobron) [inline test/PizzeriaTest.cpp:102]: price is not checked if it is equal to expected. Please calculate expected as sum of prices of each pizzas. No magic values.

**Review by ziobron — CHANGES_REQUESTED** [review]

**Comment by DamiXioR** [comment]
I have made changes. Hope it's allright now.

**Review by ziobron — APPROVED** [review]

**Comment by ziobron** [comment]
{ "requirement_ids" : [3524, 3525, 3526] }
10 XP za TimeMock w Pizzas (3524)
15 XP za nowy przypadek testowy w Pizzas (3525)
5 XP za workflow w Pizzas (3526)

**Pull request closed without merging**